File: Java/_458_Poor_Pigs/Solution.java
class Solution {
    public int poorPigs(int buckets, int minutesToDie, int minutesToTest) {
        if (buckets <= 1)
            return 0;
        int side = minutesToTest / minutesToDie + 1;
        int pigs = 1;
        int num = side;
        while (num < buckets) {
            num = num * side;
            pigs++;
        }
        return pigs;
    }
}

File: Python/_458_Poor_Pigs/solution.py
class Solution(object):
    def poorPigs(self, buckets, minutesToDie, minutesToTest):
        """
        :type buckets: int
        :type minutesToDie: int
        :type minutesToTest: int
        :rtype: int
        """
        if buckets <= 1:
            return 0
        side = minutesToTest // minutesToDie + 1
        pigs = 1
        num = side
        while num < buckets:
            num = num * side
            pigs += 1
        return pigs
